docs/jsonapi-spec.md

```diff
@@ -591,6 +628,102 @@ See [Multi-Document Failure Considerations](#multi-document-failure-consideratio
+
+<find-one-and-replace-command-option> ::= (<returnDocument>)*
+```
+
+*Sample:*
+
+```json5
+// Replaces a single document based on the specified filter and sort. Returns a document in response.
+{ "findOneAndReplace" : 
+    { "filter" : {<filter-clause>}, 
+      "sort" : {<sort-clause>}, 
+      "replacement" : {<document-to-replace>}, 
+      "options" : 
+          {"returnDocument" : "before/after"} 
+     }
+}
+```
+
+#### findOneAndReplace Command Order of Operations
+
+`findOneAndReplace` commands are processed using the following order of operation:
+
+1.  `<filter-clause>` is applied to the collection to select a single candidate document. Example:  `"filter": {"location": "London"}`.
+2.  `<sort-clause>` can be applied to the candidate document to determine its order. Example: `"sort" : ["race.start_date"]`. If no `<sort-clause>` is supplied, the document maintains its current order.
+3.  `<document-to-replace>` specifies the replacement action. Example: `"replacement": { "location": "New York", "count": 3 }`.
+
+The <replacement> document cannot specify an `_id` value that differs from the replaced document.
+
+#### findOneAndReplace Command Options
+
+`<find-one-and-replace-command-option>` is a map of key-value pairs that modify the behavior of the command. 
+
+If `returnDocument` is `before`, return the existing document. if `returnDocument` is `after`, return the replaced document. 
+
+| Option            | Type        | Description                                                                     |
+| ----------------- | ----------- | ------------------------------------------------------------------------------- |
+| `returnDocument`  | String      | Specifies which document to perform the projection on. If `"before"` the projection is performed on the document before the update is applied. If  `"after"` the document projection is from the document after the update. Defaults to `"before"`. |
+
+#### findOneAndReplace Document Failure Modes
+
+Fail Fast, a storage failure causes the command to stop processing.
+
+If the replacement document `_id` field is different from the document read from the database, the JSON API throws an error.
+
+NOTE: you can omit `_id` in the replacement document. If `_id` is in the replacement, it should be exactly equal to the `_id` in the database. But if `_id` was omitted, 
+`findOneAndReplace` will use the existing document's `_id`.
+
+#### findOneAndReplace Command Response
+
+| Response Elements | Description                                                                     |
+| ----------------- | ------------------------------------------------------------------------------- |
+| `data`            | Present with fields : `docs` only, see [findOneAndReplace Command Options](#findOneAndReplace-command-options) for controlling the projection. |
+| `status`          | Preset with fields: `upsertedId: <json-value>` only if a document was upserted. |
```

Comment:
json-value -> doc id of the document upserted?

Reply:
Thanks, updated in latest commit.